fanficfare/adapters/adapter_spiritfanfictioncom.py

```diff
@@ -0,0 +1,350 @@
+    @classmethod
+    def getSiteAbbrev(cls):
+        return 'spirit'
+    
+
+    def getStoryId(self, url):
+
+        # get storyId from url--url validation guarantees query correct
+        m = re.match(self.getSiteURLPattern(), url)
+        if m:
+            return m.group('storyId')
+        else:
+            raise exceptions.InvalidStoryURL(url, self.getSiteDomain(), self.getSiteExampleURLs())
+
+
+    def extractChapterUrlsAndMetadata(self):
+
+        data = self.get_request(self.url)
+        # use BeautifulSoup HTML parser to make everything easier to find.
+        soup = self.make_soup(data)
+        # Now go hunting for all the meta data and the chapter list.
+
+        # Title
+        title = soup.find('h1', {'class':'tituloPrincipal'})
+        self.story.setMetadata('title', title.find('strong').text.strip())
+                                
+        # Authors
+        # Find authorid and URL
+        authors = soup.findAll('span', {'class':'usuario'})
+
+        for author in authors:
+            self.story.addToList('authorId', author.find('a')['href'].split('/')[-1])
+            self.story.addToList('authorUrl', author.find('a')['href'])
+            self.story.addToList('author', author.find('strong').text.strip())
+
+        # Cover image
+        cover_img = soup.find('img', {'class':'imagemResponsiva'})
+        if cover_img:
+            self.setCoverImage(self.url, cover_img['src'])
+
+        newestChapter = None
+        self.newestChapterNum = None # save for comparing during update.
+        # Scan all chapters to find the oldest and newest
+        # Find the chapters:
+        chapters = soup.findAll('table', {'class':'listagemCapitulos espacamentoTop'})
+        for chapter in chapters:
+
+            for row in chapter.findAll('tr', {'class': 'listagem-textoBg1'}):  # Find each row with chapter info
+                # Find the chapter number
+                numero_span = row.find('div', {'class': 'numero'}).find('span').text.strip()  
+                a = row.find('a')  # Chapter link
+
+                # Datetime
+                date = a.find_next('time')['datetime']  
+                chapterDate = makeDate(date, self.dateformat).date()
+
+                # Add chapter number to title
+                chapter_title = numero_span + ". " + a.find('strong').text.strip()
+
+                self.add_chapter(chapter_title, a.get('href'), {'date': chapterDate})
+
+                if newestChapter == None or chapterDate > newestChapter:
+                    newestChapter = chapterDate
+                    self.newestChapterNum = self.story.getMetadata('numChapters')
+            
+        logger.debug('numChapters: (%s)', self.story.getMetadata('numChapters'))
+
+        # Summary
+        div_element = soup.find('div', {'class':'clearfix'})
+        summary = div_element.find('div', class_='texto')
+
+        text = summary.find('strong').next_sibling
+        full_text = ''
+        while text is not None:
```

Comment:
Is this loop just for removing `data-cfemail` `a` tag?  Because it would be shorter to find it and call `.decompose()` on it instead.

Reply:
Not only that, some users use "@" instead of "A" to avoid language filters, and we end up with fake email addresses in summary sometimes. The website protects them, so they need to be decoded.
Should be better and working fine now.